core/string/string_name.h

```diff
@@ -124,6 +124,12 @@ class StringName {
 	static StringName search(const char32_t *p_name);
 	static StringName search(const String &p_name);
 
+	_FORCE_INLINE_ int length() const {
+		return this->operator String().length();
+	}
+
+	_FORCE_INLINE_ bool is_empty() const { return length() == 0; }
```

Comment:
All empty checks I've seen before were `my_string_name == StringName()` which is pretty much `my_string_name._data == nullptr`. I just checked the `StringName` constructors and can confirm an empty string is never allowed to be stored, so a null check is enough (specially important if `length()` can be slow).

Reply:
Thank you very much for this. It also lets me remove the above String cast which I was really uncomfortable with as well.